src/PowerShellEditorServices.Host/CodeLens/CodeLensFeature.cs

```diff
@@ -67,42 +68,78 @@ internal class CodeLensFeature :
             return codeLenses;
         }
 
+        /// <summary>
+        /// The editor session context to get workspace and language server data from.
+        /// </summary>
+        private readonly EditorSession _editorSession;
+
+        /// <summary>
+        /// The json serializer instance for CodeLens object translation.
+        /// </summary>
+        private readonly JsonSerializer _jsonSerializer;
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="editorSession"></param>
+        /// <param name="jsonSerializer"></param>
+        /// <param name="logger"></param>
+        private CodeLensFeature(
+            EditorSession editorSession,
+            JsonSerializer jsonSerializer,
+            ILogger logger)
+                : base(logger)
+        {
+            _editorSession = editorSession;
+            _jsonSerializer = jsonSerializer;
+        }
+
+        /// <summary>
+        /// Get all the CodeLenses for a given script file.
+        /// </summary>
+        /// <param name="scriptFile">The PowerShell script file to get CodeLenses for.</param>
+        /// <returns>All generated CodeLenses for the given script file.</returns>
         public CodeLens[] ProvideCodeLenses(ScriptFile scriptFile)
         {
-            return
-                this.InvokeProviders(p => p.ProvideCodeLenses(scriptFile))
-                    .SelectMany(r => r)
-                    .ToArray();
+            return InvokeProviders(provider => provider.ProvideCodeLenses(scriptFile))
+                .SelectMany(codeLens => codeLens)
```

Comment:
More LINQ to remove?

Reply:
Is this the one I put back in?

Reply:
Yeah, see https://github.com/PowerShell/PowerShellEditorServices/pull/719#discussion_r208730489